components/fonts/platform/freetype/font.rs

```diff
@@ -291,8 +296,7 @@ impl PlatformFontMethods for PlatformFont {
             // scalable outlines". If it's an sfnt, we can get units_per_EM from the 'head' table
             // instead; otherwise, we don't have a unitsPerEm value so we can't compute y_scale and
             // x_scale.
-            let head = unsafe { FT_Get_Sfnt_Table(face_ptr, ft_sfnt_head) as *mut TtHeader };
-            if !head.is_null() && unsafe { (*head).table_version != 0xffff } {
+            if let Ok(head) = font_ref.clone().and_then(|font_ref| font_ref.head()) {
```

Comment:
If you use the higher-level `skrifa` crate (also from the fontations projects, builds on top of `read-fonts`), then there is a [`FontRef::metrics` method](https://docs.rs/skrifa/latest/skrifa/font/struct.FontRef.html#method.metrics) which does a lot of theses manual table accesses and calculations for you.

Reply:
Yes, it's true, but I have not been able to determine that skrifa handles unscalable fonts properly, so this change is a conservative first step. Once I have done the work to determine that the metrics we get from fontations are the same as those we get from font tables, I can use `skrifa` to get font metrics more directly.